server/app/views/admin/programs/ProgramBlockAllAspectsView.java

```diff
@@ -49,45 +53,61 @@
 import views.style.StyleUtils;
 
 /**
- * Renders a page for an admin to edit the configuration for a single block of a program.
+ * Renders a page for an admin to view or edit the configuration for a single
+ * block of an active or draft program.
  *
- * <p>Contains elements to:
+ * <p>For drafts editing this contains elements to:
  *
  * <ul>
  *   <li>Delete the block
  *   <li>Edit the name and description
  *   <li>View, add, delete and reorder questions
  *   <li>View and navigate to the visibility criteria
  * </ul>
+ *
+ * For Active program viewing it contains the same elements, but UI elements
+ * that can be used for editing.
  */
-public final class ProgramBlockEditView extends ProgramBlockView {
+public final class ProgramBlockAllAspectsView extends ProgramBlockView {
 
   private final AdminLayout layout;
   private final FeatureFlags featureFlags;
   private final boolean featureFlagOptionalQuestions;
+  private final ProgramDisplayType programDisplayType;
 
   public static final String ENUMERATOR_ID_FORM_FIELD = "enumeratorId";
   public static final String MOVE_QUESTION_POSITION_FIELD = "position";
   private static final String CREATE_BLOCK_FORM_ID = "block-create-form";
   private static final String CREATE_REPEATED_BLOCK_FORM_ID = "repeated-block-create-form";
   private static final String DELETE_BLOCK_FORM_ID = "block-delete-form";
+  private static final String NOT_YET_IMPLEMENTED_ERROR =
+    " The read only version of ProgramBlockView is not fully implemented. It should only be "
+      + "used once issue #3162 is closed.";
 
   @Inject
-  public ProgramBlockEditView(
-      AdminLayoutFactory layoutFactory, Config config, FeatureFlags featureFlags) {
+  public ProgramBlockAllAspectsView(
+    @Assisted ProgramDisplayType programViewType,
```

Comment:
I'm a little wary this is going to WAI with Play's injection model, but let me look into it more.